internal/config/config.go

```diff
@@ -602,6 +602,17 @@ var renamedSubsys = map[string]string{
 	// Add future sub-system renames
 }
 
+const ( // deprecated keys
+	apiReplicationWorkers       = "replication_workers"
+	apiReplicationFailedWorkers = "replication_failed_workers"
+)
+
+// map of subsystem to deleted keys
+var deletedSubSysKeys = map[string][]string{
+	APISubSys: {apiReplicationWorkers, apiReplicationFailedWorkers},
```

Comment:
the list of keys I think should be provided by the `internal/config/api/config.go` so that - each sub-system can have its own list of keys to deprecate.

Reply:
internal/config/api/ already imports internal/config leading to circular dependencies